src/mixing_fab_tests.cc

```diff
@@ -0,0 +1,386 @@
+    void SetInputInv(std::vector<Material::Ptr> mat);
+    
+    
+    std::vector<std::string>  GetStream_comds() { return mf_facility_->in_commods; }
+    std::vector<double> GetStream_ratio()       { return mf_facility_->mixing_ratio; }
+    std::vector<double> GetStream_capacity()    { return mf_facility_->in_buf_size; }
+    
+    std::string GetOutStream_comds()  { return mf_facility_->out_commod; }
+    double GetOutStream_capacity()    { return mf_facility_->output_buf_size; }
+    double GetThroughput()    { return mf_facility_->throughput; }
+    
+    
+    InvBuffer* GetOutPutBuffer() { return &mf_facility_->output;}
+    std::map<std::string, InvBuffer > GetStreamBuffer() { return mf_facility_->streambufs;}
+    
+    
+    
+    
+  };
+  
+  
+  void MixingFabTest::SetUp() {
+    mf_facility_ = new MixingFab(tc_.get());
+ 
+  }
+  
+  void MixingFabTest::TearDown() {
+    delete mf_facility_;
+  }
+
+  
+  void MixingFabTest::SetInputInv(std::vector< Material::Ptr > mat){
+    for (int i = 0; i < in_com.size(); i++) {
+      mf_facility_->streambufs[in_com[i]].Push(mat[i]) ;
+    }
+  }
+  
+  
+  
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  TEST_F(MixingFabTest, InitialState) {
+    // Test things about the initial state of the facility here
+    std::vector<std::string> in_com_ = { "in_c1", "in_c2", "in_c3" };
+    std::vector<double> in_frac_ = { 0.80, 0.15, 0.05 };
+    std::vector<double> in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_ratio(in_frac_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    
+    std::vector<std::string> strm_comds_ = GetStream_comds();
+    std::vector<double> strm_ratio_ = GetStream_ratio();
+    std::vector<double> strm_cap_   = GetStream_capacity();
+    
+    for (int i = 0; i < in_com.size(); i++) {
+      EXPECT_EQ(in_com[i],  strm_comds_[i]);
+      EXPECT_EQ(in_frac[i], strm_ratio_[i]);
+      EXPECT_EQ(in_cap[i],  strm_cap_[i]);
+    }
+    
+    EXPECT_EQ(out_com, GetOutStream_comds());
+    EXPECT_EQ(out_cap, GetOutStream_capacity());
+    
+    EXPECT_EQ(throughput, GetThroughput());
+      }
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  TEST_F(MixingFabTest, StreamRatio) {
+    // Test things about the mixing ratio of the facility here
+    
+    std::vector<std::string> in_com_ = { "in_c1", "in_c2", "in_c3" };
+    std::vector<double> in_cap_ = { 30, 20, 10 };
+    
+    SetStream_comds(in_com_);
+    SetStream_capacity(in_cap_);
+    
+    SetOutStream_comds("out_com");
+    SetOutStream_capacity(50);
+    
+    SetThroughput(1e200);
+    
+    mf_facility_->EnterNotify();
+    double ext_val = 1/3.;
+    std::vector<double> strm_ratio_ = GetStream_ratio();
+    for (int i = 0; i < in_com.size(); i++) {
+      EXPECT_DOUBLE_EQ(ext_val, strm_ratio_[i]);
```

Comment:
This should be its own test - and should be commented as checking that ratios correctly default to 1/N.